TodoApp/Src/TodoApp.Contracts/Enums/Lifecycle.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace TodoApp.Contracts.Enums
```

Comment:
The `IDependencyContainer`, `IBootstrap`, `DependencyResolutionFailedException` and `Lifecycle` have quite a lot in common. While the bootsrap could technically be held alone, others should be placed in since namespace, since they are often, if not always, used together.

Have a look at your exceptions folder - how much does the two have in common (despite being derived from the same class)? How often will you see them together? 
It is often better, I believe, to couple classes feature-wise (or use-wise if you will) rather than type-wise…

From this point of view, the `IDateTimeWrapper` and `IGuidWrapper` have a bit more in common - they are not related to any particular feature or part of your system, yet do their thing in similar fashion - than the `IConnectionStringWrapper`.
The `IConnectionStringWrapper` is quite single-purposed and related to `Repositories` and I would even dare argue, it is not a wrapper in the same manner as the two mentioned before, even while it does a quite similar thing (I did not complain about the `DefaultConnectionString` being a property since it truly represent a cached, unchanging value though).
Have a look at [`IOptions` pattern in Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.1) and see for yourself that it has more in common with `IRouteNames` than the other wrappers it is in the folder and namespace with (yet it still belongs even closer to `Repositories` I believe).

To make my point complete, what connects the `IItem*Service` is the `Item` - kind-of feature- or usage-wise attribute they all posses. They could, under other circumstances, be nested in a `Items`-ish subfolder, yet in this case, it is not necessary since there are no other services.